dev/buildtool/gradle_support.py

```diff
@@ -0,0 +1,452 @@
+    return failed_task, first_line
+
+  def extract_failure_summary(self, retcode, output):
+    """Determine the top level failure message from gradle."""
+    if not retcode:
+      return None, None
+    task, first_line = self.__extract_task_failure(output)
+    if task is not None:
+      return task, first_line
+    errno_match = re.search("error=", output)
+    if errno_match:
+      return 'unknown-task', errno_match.group(0)
+    return 'unknown-task', 'unknown-error'
+
+  def determine_labels(self, retcode, output):
+    """Determine label bindings for this outcome."""
+    labels = {
+        'repository': self.__name,
+        'context': self.__context,
+        'success': retcode == 0,
+        'failed_task': '',
+        'failed_by': '',
+        'failed_reason': ''
+    }
+    if retcode == 0:
+      return labels
+
+    failed_task, summary_line = self.extract_failure_summary(retcode, output)
+    self.update_failure_cause(labels, failed_task, summary_line)
+
+    return labels
+
+  def __update_http_failure_cause(self, labels, summary_line):
+    """Figure out the labels from the failure summary line."""
+    # This is an example of an error (wrapped from one to multiple lines
+    # for readability here. The ...jar was a long path to the bintray jar.
+    #
+    # > Could not upload to \
+    # 'https://api.bintray.com/content/...jar': HTTP/1.1 409 Conflict \
+    # [message:Unable to upload files: An artifact with the path \
+    # 'com/netflix/spinnaker/echo/echo-core/1.542.0/echo-core-1.542.0.jar' \
+    # already exists]
+
+    summary_match = re.match(
+        r"[> ]*(.*?)'(.+?)': HTTP/[0-9\.]+ ([0-9]{3}) ([\w]+)(.*)",
+        summary_line)
+    if not summary_match:
+      return False
+
+    # abstract = summary_match.group(1)
+    thing = summary_match.group(2)
+    abstract_http = summary_match.group(3)
+    # abstract_category = summary_match.group(4)
+    if thing.find('bintray.com') >= 0:
+      logging.debug('Counting another bintray failure.')
+      labels['failed_by'] = 'bintray'
+    labels['failed_reason'] = abstract_http
+    return True
+
+  def __update_error_failure_cause(self, labels, summary_line):
+    errno_match = re.match(
+        r".*error='(.+?)' \(errno=(\d+)\).*",
+        summary_line)
+    if errno_match:
+      cause = errno_match.group(1)
+      errno = errno_match.group(2)
+      logging.debug('FAILED errno=%s cause="%s"'
+                    '\nSUMMARY=%s', errno, cause, summary_line)
+      labels['failed_by'] = 'other_runtime'
+      labels['failed_reason'] = cause
+
+    return False
+
+  def update_failure_cause(self, labels, failed_task, summary_line):
+    """Figure out the labels from the failure summary line."""
+
+    labels['failed_task'] = failed_task
+    labels['failed_by'] = 'unknown'
+    labels['failed_reason'] = 'unknown'
+    if self.__update_http_failure_cause(labels, summary_line):
+      return
+    if self.__update_error_failure_cause(labels, summary_line):
```

Comment:
Seems like this function would never return `True`, so we'd never hit this case.